code/modules/mechs/components/_components.dm

```diff
@@ -101,27 +109,91 @@
 	to_chat(user, SPAN_NOTICE(" - Integrity: <b>[round((((max_damage - total_damage) / max_damage)) * 100)]%</b>" ))
 
 /obj/item/mech_component/attackby(obj/item/I, mob/living/user)
-	if(I.use_tool(user, src, WORKTIME_INSTANT, QUALITY_SCREW_DRIVING, FAILCHANCE_ZERO))
-		if(contents.len)
-			//Filter non movables
-			var/list/valid_contents = list()
-			for(var/atom/movable/A in contents)
-				if(!A.anchored)
-					valid_contents += A
-			if(!valid_contents.len)
-				return
-			var/obj/item/removed = pick(valid_contents)
-			if(!(removed in contents))
-				return
-			if(eject_item(removed, user))
-				update_components()
-		else
-			to_chat(user, SPAN_WARNING("There is nothing to remove."))
+	var/list/usable_qualities = list(QUALITY_PULSING, QUALITY_BOLT_TURNING, QUALITY_WELDING, QUALITY_PRYING, QUALITY_SCREW_DRIVING)
+	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+
+	switch(tool_type)
+		if(QUALITY_SCREW_DRIVING)
+			if(contents.len)
+				//Filter non movables
+				var/list/valid_contents = list()
+				for(var/atom/movable/A in contents)
+					if(!A.anchored)
+						valid_contents += A
+				if(!valid_contents.len)
+					return
+				var/obj/item/removed = pick(valid_contents)
```

Comment:
```suggestion
				var/obj/item/removed = input("Which component would you like to remove?","Removing components") in valid_contents
```
We can ask the user which one they want to remove instead of removing a random one

Reply:
oh god.. that's going to add a lot of strain to have to click thru a menu each time you want to add/remove parts. Especially if you want to remove all the parts which you can c urrently do by clicking the chassis until all parts are gone (there is no delay.) Can it be a switch depending on if you're in harm/hurt intent at least

Reply:
What about using the radial menu like how modular computers do it? Less intrusive than the byond menu and still lets players pick.

Reply:
Sounds out of scope since I didn't code it originally, I'm just moving the parts from premade > "the parts" directory but I assume y'all don't want it TM'd until next week so... sure, I'll see

Reply:
Ah, I didn't realize that. If it's not your code, it may be best to leave it and address it in a future PR.

Reply:
> 

If you feel it's better without it, no problem for me. I never used borg that much. If it's instantaneous to remove/add then it shouldn't be a problem if people remove parts they did not want to.

Reply:
🤷
i dunno it might feel different for some people who aren't sterotypical ''gamers'' I guess. I'll look into making it some sort of toggle for people who prefer menus but it's probably out of scope for this PR